protocol/peer/peer.go

```diff
@@ -87,6 +93,25 @@ func (p *Peer) UpdateFieldsFromMsgVersion(msg *wire.MsgVersion, peerID uint32) {
 	p.subnetworkID = msg.SubnetworkID
 }
 
+// SetPingPending sets the ping state of the peer to 'pending'
+func (p *Peer) SetPingPending(nonce uint64) {
+	p.pingLock.Lock()
+	defer p.pingLock.Unlock()
+
+	p.lastPingNonce = nonce
+	p.lastPingTime = time.Now()
+}
+
+// SetPingIdle sets the ping state of the peer to 'idle'
+func (p *Peer) SetPingIdle() {
+	p.pingLock.Lock()
+	defer p.pingLock.Unlock()
+
+	p.lastPingNonce = 0
+	p.lastPingMicros = time.Since(p.lastPingTime).Nanoseconds() / 1000
```

Comment:
Better use the mstime package to compute micoseconds

Reply:
Microseconds are beyond the resolution that mstime provides, I'm afraid.

Reply:
Oh, right. It's milli, you need micro

Reply:
We can call the package μstime.
I know it has been your dream since ever to have μ in our codebase.